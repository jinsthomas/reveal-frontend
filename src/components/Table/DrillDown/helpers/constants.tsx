/** Constants */
import React from 'react';

export const ID = 'id';
export type ID = typeof ID;

export const PARENT_ID = 'parent_id';
export type PARENT_ID = typeof PARENT_ID;

export const ROOT_PARENT_ID = null;
export type ROOT_PARENT_ID = typeof ROOT_PARENT_ID;

export const LINKER_ITEM_CSS_CLASS = 'dd-linker-item';
export type LINKER_ITEM_CSS_CLASS = typeof LINKER_ITEM_CSS_CLASS;

export const CLICKABLE_CSS_CLASS = 'dd-linker-item dd-clickable';
export type CLICKABLE_CSS_CLASS = typeof CLICKABLE_CSS_CLASS;

export const CARET_CSS_CLASS = 'dd-caret';
export type CARET_CSS_CLASS = typeof CARET_CSS_CLASS;

export const CARET_SPAN = <span className={CARET_CSS_CLASS}>&nbsp;&#9660;</span>;
export type CARET_SPAN = typeof CARET_SPAN;

export const DEFAULT_ROW_HEIGHT = '2em';
export type DEFAULT_ROW_HEIGHT = typeof DEFAULT_ROW_HEIGHT;
