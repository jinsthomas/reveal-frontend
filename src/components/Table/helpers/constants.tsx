/** Constants */
import React from 'react';
export const ID = 'id';
export type ID = typeof ID;

export const PARENT_ID = 'parent_id';
export type PARENT_ID = typeof PARENT_ID;

export const ROOT_PARENT_ID = null;
export type ROOT_PARENT_ID = typeof ROOT_PARENT_ID;

export const LINKER_ITEM_CSS_CLASS = 'dd-linker-item';
export type LINKER_ITEM_CSS_CLASS = typeof LINKER_ITEM_CSS_CLASS;

export const CLICKABLE_CSS_CLASS = 'dd-linker-item dd-clickable';
export type CLICKABLE_CSS_CLASS = typeof CLICKABLE_CSS_CLASS;

export const CARET_CSS_CLASS = 'dd-caret';
export type CARET_CSS_CLASS = typeof CARET_CSS_CLASS;

export const CARET = <span className={CARET_CSS_CLASS}>&nbsp;&#9660;</span>;
export type CARET = typeof CARET;

export const ROWS_TO_DISPLAY = 'Rows to display';
export type ROWS_TO_DISPLAY = typeof ROWS_TO_DISPLAY;

export const PAGE = 'page';
export type PAGE = typeof PAGE;

export const OF = 'of';
export type OF = typeof OF;
