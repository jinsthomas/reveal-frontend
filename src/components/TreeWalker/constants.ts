export const ACTIVE = 'Active';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
