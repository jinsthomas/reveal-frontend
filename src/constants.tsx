// env vars
export const WEBSITE_NAME = process.env.REACT_APP_WEBSITE_NAME;
export type WEBSITE_NAME = typeof WEBSITE_NAME;

// strings
export const LOCATION = 'Location';
export type LOCATION = typeof LOCATION;
export const PROVINCE = 'Province';
export type PROVINCE = typeof PROVINCE;
export const HOME = 'Home';
export type HOME = typeof HOME;
export const FOCUS_INVESTIGATION = 'Focus Investigation';
export type FOCUS_INVESTIGATION = typeof FOCUS_INVESTIGATION;
export const FOCUS_INVESTIGATIONS = 'Focus Investigations';
export type FOCUS_INVESTIGATIONS = typeof FOCUS_INVESTIGATIONS;

// internal urls
export const HOME_URL = '/';
export type HOME_URL = typeof HOME_URL;
export const IRS_URL = '/irs';
export type IRS_URL = typeof IRS_URL;
export const FI_URL = '/focus-investigation';
export type FI_URL = typeof FI_URL;
export const FI_HISTORICAL_URL = '/focus-investigation/historical';
export type FI_HISTORICAL_URL = typeof FI_HISTORICAL_URL;

// colors
export const GREEN = 'Green';
export type GREEN = typeof GREEN;
export const YELLOW = 'Yellow';
export type YELLOW = typeof YELLOW;
export const ORANGE = 'Orange';
export type ORANGE = typeof ORANGE;
export const RED = 'Red';
export type RED = typeof RED;

// thresholds
export const GREEN_THRESHOLD = 0.9;
export type GREEN_THRESHOLD = typeof GREEN_THRESHOLD;
export const YELLOW_THRESHOLD = 0.2;
export type YELLOW_THRESHOLD = typeof YELLOW_THRESHOLD;
export const ORANGE_THRESHOLD = 0.8;
export type ORANGE_THRESHOLD = typeof ORANGE_THRESHOLD;

// 1-3-7 thresholds
export const THRESHOLD_137 = 0;
