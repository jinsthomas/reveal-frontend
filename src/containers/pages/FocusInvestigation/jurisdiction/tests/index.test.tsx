import reducerRegistry from '@onaio/redux-reducer-registry';
import { shallow } from 'enzyme';
import { createBrowserHistory } from 'history';
import React from 'react';
import { MemoryRouter } from 'react-router';
import { FI_SINGLE_URL } from '../../../../../constants';
import plansReducer, { reducerName as plansReducerName } from '../../../../../store/ducks/plans';
import { defaultActiveFIProps, FIJurisdiction } from '../index';

jest.mock('../../../../../configs/env');

/** register the jurisdictions reducer */
reducerRegistry.register(plansReducerName, plansReducer);

const history = createBrowserHistory();

describe('containers/FocusInvestigation/Jurisdiction', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    const mock: any = jest.fn();

    const props = {
      ...defaultActiveFIProps,
      history,
      location: mock,
      match: {
        isExact: true,
        params: { jurisdictionId: 'fixtures.plan1.id' },
        path: `${FI_SINGLE_URL}/:jurisdictionId`,
        url: `${FI_SINGLE_URL}/fixtures.plan1.id`,
      },
    };
    shallow(
      <MemoryRouter>
        <FIJurisdiction {...props} />
      </MemoryRouter>
    );
  });
});
