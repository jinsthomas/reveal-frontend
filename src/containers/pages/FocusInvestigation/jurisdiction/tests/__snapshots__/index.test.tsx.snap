// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/FocusInvestigation/Jurisdiction renders correctly: HeaderBreadcrumb li a 1`] = `
Array [
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>,
  <a
    href="/focus-investigation"
    onClick={[Function]}
  >
    Focus Investigations
  </a>,
  <a
    href="/focus-investigation/filter/61707fc2-c6ac-4112-a8d6-2a4861958396/236ca3fb-1b74-5028-a0c8-ab954bb28055"
    onClick={[Function]}
  >
    Trat 2
  </a>,
  <a
    href="/focus-investigation/filter/8a8e4987-95a9-4b2e-b746-3c9c3eec15c5/236ca3fb-1b74-5028-a0c8-ab954bb28055"
    onClick={[Function]}
  >
    Mueng Trat District 2
  </a>,
  <a
    href="/focus-investigation/filter/735a94f9-6ed2-4bcd-b242-8c2b5f714bee/236ca3fb-1b74-5028-a0c8-ab954bb28055"
    onClick={[Function]}
  >
    Laem Klat Canton 2
  </a>,
  <a
    href="/focus-investigation/filter/a14c6253-2d3d-4bcc-8c49-35f9e44c66e1/236ca3fb-1b74-5028-a0c8-ab954bb28055"
    onClick={[Function]}
  >
    Laem Klat Moo 8
  </a>,
]
`;

exports[`containers/FocusInvestigation/Jurisdiction renders correctly: JurisdictionMap 1`] = `
<JurisdictionMap
  callback={[Function]}
  jurisdictionId="450fc15b-5bd2-468a-927a-49cb10d3bcac"
>
  <div>
    I love oov
  </div>
</JurisdictionMap>
`;

exports[`containers/FocusInvestigation/Jurisdiction renders correctly: Link to create new plans 1`] = `
<a
  className="focus-investigation btn btn-primary float-right mt-5"
  href="/plans/new"
  onClick={[Function]}
>
  Add Focus Investigation
</a>
`;

exports[`containers/FocusInvestigation/Jurisdiction renders correctly: Link to create new plans 2`] = `
<a
  className="focus-investigation btn btn-primary float-right mt-5"
  href="/plans/new"
  onClick={[Function]}
>
  Add Focus Investigation
</a>
`;

exports[`containers/FocusInvestigation/Jurisdiction renders correctly: focus-area-info-section 1`] = `
<dl
  className="row mt-3"
>
  <dt
    className="col-4"
  >
    Province
  </dt>
  <dd
    className="col-8"
  >
    Trat 2
  </dd>
  <dt
    className="col-4"
  >
    District
  </dt>
  <dd
    className="col-8"
  >
    Mueng Trat District 2
  </dd>
  <dt
    className="col-4"
  >
    Canton
  </dt>
  <dd
    className="col-8"
  >
    Laem Klat Canton 2
  </dd>
  <dt
    className="col-4"
  >
    FI Status
  </dt>
  <dd
    className="col-8"
  >
    A1
  </dd>
  <dt
    className="col-4"
  >
    FI Reason
  </dt>
  <dd
    className="col-8"
  >
    Case Triggered
  </dd>
</dl>
`;
